src/System.Private.ServiceModel/src/System/IdentityModel/Claims/WindowsClaimSet.cs

```diff
@@ -1,16 +1,349 @@
+                ClaimTypes.Name == claimType;
+        }
+
+        // Note: null string represents any.
+        public override IEnumerable<Claim> FindClaims(string claimType, string right)
+        {
+            ThrowIfDisposed();
+            if (!SupportedClaimType(claimType) || !ClaimSet.SupportedRight(right))
+            {
+                yield break;
+            }
+            else if (_claims == null && (ClaimTypes.Sid == claimType || ClaimTypes.DenyOnlySid == claimType))
+            {
+                if (ClaimTypes.Sid == claimType)
+                {
+                    if (right == null || Rights.Identity == right)
+                    {
+                        yield return new Claim(ClaimTypes.Sid, _windowsIdentity.User, Rights.Identity);
+                    }
+                }
+
+                if (right == null || Rights.PossessProperty == right)
+                {
+                    Claim sid;
+                    if (TryCreateWindowsSidClaim(_windowsIdentity, out sid))
+                    {
+                        if (claimType == sid.ClaimType)
+                        {
+                            yield return sid;
+                        }
+                    }
+                }
+
+                if (_includeWindowsGroups && (right == null || Rights.PossessProperty == right))
+                {
+                    // Not sure yet if GroupSidClaimCollections are necessary in .NET Core
+                    //for (int i = 0; i < Groups.Count; ++i)
+                    //{
+                    //    Claim sid = Groups[i];
+                    //    if (claimType == sid.ClaimType)
+                    //    {
+                    //        yield return sid;
+                    //    }
+                    //}
+                }
+            }
+            else
+            {
+                EnsureClaims();
+
+                bool anyClaimType = (claimType == null);
+                bool anyRight = (right == null);
+
+                for (int i = 0; i < _claims.Count; ++i)
+                {
+                    Claim claim = _claims[i];
+                    if ((claim != null) &&
+                        (anyClaimType || claimType == claim.ClaimType) &&
+                        (anyRight || right == claim.Right))
+                    {
+                        yield return claim;
+                    }
+                }
+            }
+        }
+
+        public override IEnumerator<Claim> GetEnumerator()
+        {
+            ThrowIfDisposed();
+            EnsureClaims();
+            return _claims.GetEnumerator();
+        }
+
+        public override string ToString()
+        {
+            return _disposed ? base.ToString() : SecurityUtils.ClaimSetToString(this);
+        }
+
+        [Fx.Tag.SecurityNote(Critical = "Uses critical type SafeHGlobalHandle.",
+            Safe = "Performs a Demand for full trust.")]
+        [SecuritySafeCritical]
```

Comment:
I don't think we need SecuritySafeCritical as we're not supporting partial trust.